proto/warrior.proto

```diff
@@ -57,12 +57,62 @@ message WarriorTalents {
 
 message Warrior {
     message Rotation {
+		enum Type {
+			ArmsSlam = 0;
+			ArmsDW = 1;
+			Fury = 2;
+			General = 3;
+		}
+
+		Type type = 1;
+
+		message ArmsSlamRotation {
+
+			bool use_slam_exec = 1;
+			double slam_latency = 2;
+			bool use_ms_exec = 3;
+		}
+		ArmsSlamRotation arms_slam = 2;
+
+		message ArmsDWRotation {
+
+			bool use_ms_exec = 1;
+		}
+		ArmsDWRotation  arms_dw = 3;
+
+		message FuryRotation {
+			enum PrimaryInstant {
+				Bloodthirst = 0;
+				Whirlwind = 1;
+			}
+			PrimaryInstant primary_instant = 1;
+
+			bool use_bt_exec = 2;
+			double rampage_cd_tresh = 10; // only when using rampage
+
+		}
+		FuryRotation fury = 4;
+
+		message GeneralRotation {
+			bool use_ww_exec = 1;
+			bool use_hs_exec = 2;
+			double hs_rage_thresh = 3;
+			bool use_overpower = 4;
+			double overpower_rage_thresh = 5;
+			bool use_hamstring = 6;
+			double hamstring_rage_thresh = 7;
```

Comment:
These are the variables I referred to before, you can remove the 'GeneralRotation' message and just have these within the Rotation message.

Also, I'm seeing the naming pattern '_thresh' and 'use_xx_exec' a lot, could we make them '_threshold' and 'use_xx_during_execute' instead? I know its a little verbose but it helps a lot if a non-warrior has to read your code.

Reply:
Done in a21327c1.